### ISSUE TITLE
Streaming server

### ISSUE THREAD

**Issue by karantin2020** [h2o/h2o#853]
Hello.
Could you help please. What is a way in h2o to stream some set of data? 
For example:
```c
{
	{-97, 37.819929, -122.478255},
	{1899, 39.096849, -120.032351},
	{2619, 37.865101, -119.538329},
	{42, 33.812092, -117.918974},
	{15, 37.77493, -122.419416},
}
```
How to stream this array struct by struct?
Is it possible to make audio/video streaming server with h2o?

**Comment by kazuho** [comment]
> What is a way in h2o to stream some set of data?

Call `h2o_send` with its final argument (i.e. `is_final`) set to zero.

The contract that should be obeyed by a handler is:
* the _last_ call to `h2o_send` must have its `is_final` flag set to one
* if the `is_final` flag passed to `h2o_send` is zero, then
 * the handler should register the `proceed` callback as well
 * must not call `h2o_send` next time, until the `proceed` callback is called

In other words, the call sequence should be:

1. h2o_send(is_final=0)
2. proceed callback
3. h2o_send(is_final=0)
4. proceed callback
5. ...
6. h2o_send(is_final=1) // proceed callback never gets called after this

Many handlers (e.g. file, proxy, fastcgi) implements this.

**Comment by karantin2020** [comment]
Thanks
